validator_client/src/http_api/mod.rs

```diff
@@ -1235,29 +1268,10 @@ where
     F: FnOnce() -> Result<T, warp::Rejection> + Send + 'static,
     T: Serialize + Send + 'static,
 {
-    warp_utils::task::blocking_task(func)
-        .await
-        .map(|func_output| {
-            let mut response = match serde_json::to_vec(&func_output) {
-                Ok(body) => {
-                    let mut res = Response::new(body);
-                    res.headers_mut()
-                        .insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
-                    res
-                }
-                Err(_) => Response::builder()
-                    .status(StatusCode::INTERNAL_SERVER_ERROR)
-                    .body(vec![])
-                    .expect("can produce simple response from static values"),
-            };
-
-            let body: &Vec<u8> = response.body();
-            let signature = signer(body);
-            let header_value =
-                HeaderValue::from_str(&signature).expect("hash can be encoded as header");
-
-            response.headers_mut().append("Signature", header_value);
-
-            response
-        })
+    let result = warp_utils::task::blocking_task(func).await;
+    let conversion = convert_rejection(result).await; // It handles the rejection
```

Comment:
I'm not following the logic here, `convert_rejection` isn't available in the scope, perhaps you're trying to call the `convert_with_header` you defined above?

Reply:
Right, I’ve already modified it but haven’t uploaded the last commit. I’m using ‘convert_with_header’ instead of ‘convert_rejection

Reply:
I've made significant progress. I'll check tomorrow to see if only the `.then` is causing the failure, and then I'll modify the handlers that call `blocking_signed_task` :)